sunxi: add support for Kaiser Baas KBA03042

photonican opened the pull request:
Specification
-------------
Full Name : Kaiser Baas Smart Media Player KBA03042
SoC       : Allwinner A20 sun7i dual-core ARM Cortex-A7
CPU Mhz   : 960MHz (1.2Ghz Vendor spec. Likely too high as no cooling)
RAM       : DDR3 SDRAM 1024Mbytes (4 x SK Hynix H5TQ2G83CFR-H9C)
RAM MHz   : 480MHz clock (960MT/s)
Flash     : 4GB MLC NAND Flash (Micron 29F32G08CBACA)
SD Card   : 1x SDXC Slot
WLAN      : 802.11 b/g/n 2.4GHz (Realtek RTL8188EUS) ** DRIVER NOT WORKING
Ethernet  : 10/100 Mbps x1 LAN RJ45 (Realtek RTL8201CP)
UART      : RX & TX pads on PCB adjacent to the flash chip (115200, 8N1)
USB Ports : 2x USB 2.0 (1x Standard + 1x OTG) both wired to USB-A sockets
LEDs      : 1x Power (Blue), gpio PH20
            1x Standby (Red), Remote power button long press -> Standby
            1x LAN activity (Orange)
            1x LAN link (Green)
Power     : 5VDC, 2A
AV        : HDMI, Composite 3.5mm, S/PDIF
IR        : 1x IR Receiver

Installation
------------
Write image to SD card. Insert SD card and power on.

map-b commented on target/linux/sunxi/image/cortexa7.mk:
```diff
@@ -70,6 +70,22 @@ define Device/friendlyarm_zeropi
 endef
 TARGET_DEVICES += friendlyarm_zeropi
 
+define Device/kaiserbaas_kba03042
```
To match the DTS compatible and avoid the need to add SUPPORTED_DEVICES, `kaiserbaas_kaiserbaas-kba03042`

abajk commented:
Consider sending dts upstream.

photonican commented on package/boot/uboot-sunxi/src/arch/arm/dts/sun7i-a20-kaiserbaas-kba03042.dts:
I've moved the dts & also the defconfig file into `package/boot/uboot-sunxi/patches/262-add-kaiserbaas-kba03042.patch`.

photonican committed:
sunxi: add support for Kaiser Baas KBA03042

Specification
-------------
Full Name : Kaiser Baas Smart Media Player KBA03042
SoC       : Allwinner A20 sun7i dual-core ARM Cortex-A7
CPU Mhz   : 960MHz (1.2Ghz Vendor spec. Likely too high as no cooling)
RAM       : DDR3 SDRAM 1024Mbytes (4 x SK Hynix H5TQ2G83CFR-H9C)
RAM MHz   : 480MHz clock (960MT/s)
Flash     : 4GB MLC NAND Flash (Micron 29F32G08CBACA)
SD Card   : 1x SDXC Slot
WLAN      : 802.11 b/g/n 2.4GHz (Realtek RTL8188EUS) ** DRIVER NOT WORKING
Ethernet  : 10/100 Mbps x1 LAN RJ45 (Realtek RTL8201CP)
UART      : RX & TX pads on PCB adjacent to the flash chip (115200, 8N1)
USB Ports : 2x USB 2.0 (1x Standard + 1x OTG) both wired to USB-A sockets
LEDs front: 1x Power (Blue), gpio PH20,
            1x Standby (Red), Remote power button long press -> Standby
LEDs back : 1x LAN activity (Orange) & 1x LAN link (Green)
Power     : 5VDC, 2A
AV        : HDMI, Composite 3.5mm, S/PDIF
IR        : 1x IR Receiver

Installation
------------
Write image to SD card. Insert SD card and power on.

Signed-off-by: Jianni Minicozzi <photonican@gmail.com>

map-b commented on package/boot/uboot-sunxi/patches/262-add-kaiserbaas-kba03042.patch:
```diff
@@ -0,0 +1,255 @@
++
++/ {
++	model = "Kaiser Baas Smart Media Player KBA03042";
++	compatible = "kaiserbaas,kba03042", "allwinner,sun7i-a20";
```
```suggestion
+	compatible = "kaiserbaas,kaiserbaas-kba03042", "allwinner,sun7i-a20";
```
